File: benchmarks/reference/matrix_mul.py

import time

if __name__ == "__main__":
    start = time.time()
    size = 50
    a = []
    b = []
    
    for i in range(size):
        row_a = []
        row_b = []
        for j in range(size):
            row_a.append(i + j)
            row_b.append(i - j)
        a.append(row_a)
        b.append(row_b)
    
    c = []
    for i in range(size):
        row_c = []
        for j in range(size):
            sum_val = 0
            for k in range(size):
                sum_val += a[i][k] * b[k][j]
            row_c.append(sum_val)
        c.append(row_c)
        
    print(f"Matrix[0][0]: {c[0][0]}")
    elapsed = time.time() - start
    print(f"Time: {elapsed:.6f}")

File: benchmarks/reference/matrix_mul.js

const start = process.hrtime();
const size = 50;
let a = [];
let b = [];

for (let i = 0; i < size; i++) {
    let row_a = [];
    let row_b = [];
    for (let j = 0; j < size; j++) {
        row_a.push(i + j);
        row_b.push(i - j);
    }
    a.push(row_a);
    b.push(row_b);
}

let c = [];
for (let i = 0; i < size; i++) {
    let row_c = [];
    for (let j = 0; j < size; j++) {
        let sum = 0;
        for (let k = 0; k < size; k++) {
            sum += a[i][k] * b[k][j];
        }
        row_c.push(sum);
    }
    c.push(row_c);
}

console.log("Matrix[0][0]: " + c[0][0]);
const diff = process.hrtime(start);
const elapsed = diff[0] + diff[1] / 1e9;
console.log("Time: " + elapsed.toFixed(6));
